### PR TITLE
Ruelvls 12 vpeter

### PR DESCRIPTION
**Pull request by v-peter**

### CODE REVIEW

File: src/main/java/com/greenfox/lvlup/LvlupApplication.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
+
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/LvlupApplication.java:7]: but why?

File: src/main/java/com/greenfox/lvlup/controller/BadgeController.java [hunk 2]
```diff
@@ -9,16 +9,20 @@
...
   @GetMapping(value = "/badges", produces = MediaType.APPLICATION_JSON_VALUE)
-  public ResponseEntity<?> showBadges(@RequestHeader(value = "userTokenAuth", required = false) String token) throws Exception {
+  public ResponseEntity<Object> showBadges(@RequestHeader(value = "userTokenAuth", required = false) String token) throws GeneralException {
     if (token == null || token.equals("")) {
       throw new GeneralException("Unauthorized", HttpStatus.UNAUTHORIZED);
-    } else return new ResponseEntity<>(testUserBadges, HttpStatus.OK);
+    } else return new ResponseEntity<>(this.testUserBadges, HttpStatus.OK);
   }
+
+
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/controller/BadgeController.java]: some new lines again :(

File: src/main/java/com/greenfox/lvlup/controller/BadgeController.java [hunk 3]
```diff
@@ -9,16 +9,20 @@
...
-
   private UserBadgeSetDTO testUserBadges = new UserBadgeSetDTO();
 
   @GetMapping(value = "/badges", produces = MediaType.APPLICATION_JSON_VALUE)
-  public ResponseEntity<?> showBadges(@RequestHeader(value = "userTokenAuth", required = false) String token) throws Exception {
+  public ResponseEntity<Object> showBadges(@RequestHeader(value = "userTokenAuth", required = false) String token) throws GeneralException {
     if (token == null || token.equals("")) {
       throw new GeneralException("Unauthorized", HttpStatus.UNAUTHORIZED);
-    } else return new ResponseEntity<>(testUserBadges, HttpStatus.OK);
+    } else return new ResponseEntity<>(this.testUserBadges, HttpStatus.OK);
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/controller/BadgeController.java]: Are you sure you have to use "this" keyword there?

Comment (drexeverest) [reply src/main/java/com/greenfox/lvlup/controller/BadgeController.java]: You don't need that else, just write it simple after the if

File: src/main/java/com/greenfox/lvlup/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController("/user")
+public class UserController {
+  @Autowired
+  UserService service;
+
+  @GetMapping
+  public Object liseUsers(@RequestParam long id, @RequestHeader String userTokenAuth) {
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/controller/UserController.java]: liseUser, isn't telling me much, list or get would be better

Comment (drexeverest) [reply src/main/java/com/greenfox/lvlup/controller/UserController.java]: be more specific, without Object use something like List of users

File: src/main/java/com/greenfox/lvlup/model/entity/BadgeLevel.java [hunk 5]
```diff
@@ -32,19 +27,14 @@ public BadgeLevel() {
     }
 
     public BadgeLevel(int level, String description, Badge badge, User... holders) {
+
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/model/entity/BadgeLevel.java]: not necessary new line

File: src/main/java/com/greenfox/lvlup/model/entity/BadgeLevel.java [hunk 6]
```diff
@@ -32,19 +27,14 @@ public BadgeLevel() {
...
         this.level = level;
         this.description = description;
         this.badge = badge;
         this.holders = Stream.of(holders).collect(Collectors.toSet());
         this.holders.forEach(x -> x.getBadgeLevels().add(this));
     }
 
-    public BadgeLevel(int level, String description, Badge badge) {
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/model/entity/BadgeLevel.java]: necessary constructor

File: src/main/java/com/greenfox/lvlup/repositrory/BadgeRepository.java [hunk 7]
```diff
@@ -1,12 +1,9 @@
...
 import com.greenfox.lvlup.model.entity.Badge;
-import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
-import java.util.List;
-
 @Repository
-public interface BadgeRepository extends CrudRepository <Badge, Long> {
-    List<Badge> findAll();
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/repositrory/BadgeRepository.java:11]: we need this method

File: src/main/java/com/greenfox/lvlup/service/UserService.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  public UserDto getUserDetailsById(long id) {
+    User user = this.repo.findById(id).orElse(null);
+    UserDto dto = mapper.map(user, UserDto.class);
+    dto.setBadges(getUserBadgeDTOs(user));
+    return dto;
+  }
+
+  public Set<UserBadgeDTO> getUserBadgeDTOs(User user) {
+    Set<UserBadgeDTO> badgeSet = new HashSet<>();
+    for (BadgeLevel bl :
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/service/UserService.java:30]: can we have this foreach in one line plz

File: src/main/java/com/greenfox/lvlup/service/UserService.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.Set;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Service
+public class UserService {
+  @Autowired
+  UserRepository repo;
+
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/service/UserService.java:18]: Create/Use a Bean for a ModelMapper and autowire it

File: src/main/java/com/greenfox/lvlup/repositrory/LevelRepository.java [hunk 10]
```diff
@@ -1,9 +1,9 @@
 package com.greenfox.lvlup.repositrory;
 
 import com.greenfox.lvlup.model.entity.BadgeLevel;
-import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface BadgeLevelRepository extends CrudRepository<BadgeLevel, Long> {
+public interface LevelRepository extends JpaRepository<BadgeLevel, Long> {
```
Comment (drexeverest) [inline src/main/java/com/greenfox/lvlup/repositrory/LevelRepository.java]: BadgeLevelRepository would be a better name

File: src/main/resources/application-prod.properties [hunk 11]
```diff
@@ -1,7 +1,9 @@
 spring.datasource.url=${PROD_DB_URL}
 spring.datasource.username=${PROD_DB_USERNAME}
 spring.datasource.password=${PROD_DB_PASSWORD}
+
+
```
Comment (drexeverest) [inline src/main/resources/application-prod.properties]: ➖

**Review by drexeverest — CHANGES_REQUESTED** [review]

**Review by BPAndrea — APPROVED** [review]
It's ok from my side.

**Review by drexeverest — APPROVED** [review]

**Pull request merged**